lib/operation-helper.specs.js

```diff
@@ -148,6 +148,101 @@ describe('OperationHelper', function () {
         })
     })
 
+    describe('#getQueueSize', () => {
+      let opHelper, actual
+
+      context('no throttling', () => {
+        before(() => {
+          opHelper = new OperationHelper({
+              awsId: 'testAwsId',
+              awsSecret: 'testAwsSecret',
+              assocId: 'testAssocId',
+              maxRequestsPerSecond: 0
+          })
+
+          opHelper.execute('ItemSearch', {
+              'SearchIndex': 'Books',
+              'Keywords': 'harry potter',
+              'ResponseGroup': 'ItemAttributes,Offers'
+          })
+
+          actual = opHelper.getQueueSize()
+        })
+
+        it('should return -1', () => {
+            expect(actual).to.equal(-1)
+        })
+      })
+
+      context('throttling, but with an empty queue', () => {
+        before(() => {
+          opHelper = new OperationHelper({
+              awsId: 'testAwsId',
+              awsSecret: 'testAwsSecret',
+              assocId: 'testAssocId',
+              maxRequestsPerSecond: 1
+          })
+
+          actual = opHelper.getQueueSize()
+        })
+
+        it('should return 0', () => {
+            expect(actual).to.equal(0)
+        })
+      })
+
+      context('throttling with completed queue', () => {
+        before(() => {
+          opHelper = new OperationHelper({
+              awsId: 'testAwsId',
+              awsSecret: 'testAwsSecret',
+              assocId: 'testAssocId',
+              maxRequestsPerSecond: 1
+          })
+
+          opHelper.execute('ItemSearch', {
```

Comment:
Two problems here. This file contains unit tests, but you have added an integration test here. Secondly, you are not returning the promise that is returned from the execute call, therefore mocha is not actually waiting for your promise to resolve. According to Travis CI, this test is passing, but I don't know how. Possibly non-deterministic/race-condition.

I wish I had written tests for Throttler. Then you could just stub Throttler here and test that we return whatever this.throttler.getQueueSize() returns. So, one option would be to write tests for Throttler so that we can proceed with this option.

If we don't write tests for Throttler, then we have to keep the tests very close to how you have written them, but we just need to modify them so that no actual calls to amazon are made. One idea I have is to do this by stubbing this._execute to be a no-op.